### PR TITLE
Kubernetes discovery api: only consider running pods

### PR DESCRIPTION
**Pull request by fredfp**
fix for https://github.com/akka/akka-management/issues/520

### CODE REVIEW

File: discovery-kubernetes-api/src/main/scala/akka/discovery/kubernetes/PodList.scala [hunk 1]
```diff
@@ -19,7 +19,7 @@ import akka.annotation.InternalApi
 
   final case class PodSpec(containers: immutable.Seq[Container])
 
-  final case class PodStatus(podIP: Option[String])
+  final case class PodStatus(podIP: Option[String], phase: Option[String])
```
Comment (raboof) [inline discovery-kubernetes-api/src/main/scala/akka/discovery/kubernetes/PodList.scala]: Why an `Option`? Is it not guaranteed to be there?

(I guess the same could be said about the `podIP`, though)

Comment (fredfp) [reply discovery-kubernetes-api/src/main/scala/akka/discovery/kubernetes/PodList.scala]: I wondered the same and preferred to be on the safe side

**Comment by chbatey** [comment]
Thanks, this is definitely the right thing to do.

Does bootstrap work if it gets no contact points back? Thinking of initial deployment when no pods will be running.

**Comment by fredfp** [comment]
Is it even possible that no contact point is returned? I don't think so. From [pod lifecycle](https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#pod-phase), a pod will be `Running` when `The Pod has been bound to a node, and all of the Containers have been  created. At least one Container is still running, or is in the process  of starting or restarting`. As such the initial pod should at least discover itself and bootstrapping with `contact-point-discovery.required-contact-point-nr` set to `1` should work.

**Comment by chbatey** [comment]
Sorry I got confused between Running and healthy when I wrote that. LGTM

**Pull request merged**